compiler/riscv/codegen/RealRegister.hpp

```diff
@@ -57,4 +57,25 @@ inline TR::RealRegister *toRealRegister(TR::Register *r)
    return static_cast<TR::RealRegister *>(r);
    }
 
+
+// TODO: Move it to common code (OMRRealRegister.hpp)
+inline TR::RealRegister::RegNum& operator++(TR::RealRegister::RegNum& rn)
+   {
+   // TODO: Is this assert useful? Moreover, for POWER it would have to take CCRs
+   // and vector into an account...
+   TR_ASSERT_FATAL(
+         ((TR::RealRegister::FirstGPR <= rn) && (rn <= TR::RealRegister::LastGPR))
+               || ((TR::RealRegister::FirstFPR <= rn) && (rn <= TR::RealRegister::LastFPR)),
+         "Invalid value");
```

Comment:
The need for ++ operator came from desire to be able to iterate over registers using for-loop without cluttering the code with casts. Initially I wanted to guard myself against misuse, but it hard to define the meaning of it on its own (i.e, `LastGPR++` must be allowed but does not yield an architectural register) 

I'll remove it.

Side note - it'd be nice to be able to use range-based for loops for this and also instead of `FOR_EACH_*_REGISTER` macros, but it is my understanding range-based loops are banned for now due to compiler limitations.